README.md

```diff
@@ -0,0 +1,73 @@
+# Overview
+
+This is the base layer for charms that wish to take advantage of
+Apache's Bigtop framework for configuring and deploying services via
+puppet. [Including this layer][building] gives you access to a Python
+class called Bigtop, which contains various useful tools, including an
+automatically run install routine.
+
+# Usage
+
+To create a charm using this base layer, first you must include it in
+your `layer.yaml` file:
+
+```yaml
+includes: ['layer:apache-bigtop-base']
+```
+
+This will fetch the layer from [interfaces.juju.solutions][] and
+incorporate it into your charm. To use the Bigtop class, import it,
+then call its .render_site_yaml and .trigger_puppet routines in
+sequence, like so:
+
+```python
+from charms.layer.apache_bigtop_base import Bigtop
+
+# Setup arguments to pass to .render_site_yaml
+hosts = {'myservice': <host>}
+roles = ['myservice-worker', 'myservice-client']
+override = {'myservice::somevalue': <somevalue>, ...}
+
```

Comment:
I think we need to be more precise here. For examples, the roles are actually the bigtop roles that they do not follow the pattern "'myservice-worker', 'myservice-client'". Instead we should be giving some link to a list of bigtop roles like this one inferred from https://github.com/apache/bigtop/blob/master/bigtop-deploy/puppet/manifests/cluster.pp#L16 . Unfortunately I haven't found an exhaustive list of all bigtop roles. Similarly, the hosts is there to indicate some fqdn the service we deploy might need, again a list of host types that we support would add more value.